server/player/player.go

```diff
@@ -2036,6 +2043,23 @@ func (p *Player) Tick(w *world.World, current int64) {
 	}
 	p.cooldownMu.Unlock()
 
+	if w.Dimension() == world.Overworld || w.Dimension() == world.Nether {
+		if _, ok := w.Block(cube.PosFromVec3(p.Position())).(block.Portal); ok {
```

Comment:
This should check if the player's bounding box collides with any portals around it.

Reply:
What do you think is the best way to do that with the current API?

Reply:
There currently isn't anything in the API that's specifically for this, but it's still possible with some code that looks like `(*player.Player).checkOnGround`

Reply:
Changed.